contracts/external/cw-payroll/src/contract.rs

```diff
@@ -0,0 +1,1462 @@
+use std::vec;
+
+use crate::balance::*;
+use crate::error::{ContractError, GenericError};
+use crate::linking::*;
+use crate::msg::{
+    ConfigResponse, ContractResult, ExecuteMsg, InstantiateMsg, ListStreamsResponse, QueryMsg,
+    ReceiveMsg, StreamId, StreamParams, StreamResponse,
+};
+use crate::state::{
+    add_stream, remove_stream, save_stream, Config, Stream, CONFIG, STREAMS, STREAM_SEQ,
+};
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, BankMsg, Binary, Coin, CosmosMsg, Deps, DepsMut, Env, MessageInfo,
+    Order, Response, StdResult, Storage, Uint128,
+};
+use cw2::set_contract_version;
+use cw20::{Cw20CoinVerified, Cw20Contract, Cw20ExecuteMsg, Cw20ReceiveMsg};
+use cw_storage_plus::Bound;
+
+const CONTRACT_NAME: &str = "crates.io:cw-escrow-streams";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> ContractResult {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let admin = match msg.admin {
+        Some(ad) => deps.api.addr_validate(&ad)?,
+        None => info.sender,
+    };
+
+    let config = Config {
+        admin: admin.clone(),
+    };
+    CONFIG.save(deps.storage, &config)?;
+
+    STREAM_SEQ.save(deps.storage, &0u64)?;
+
+    Ok(Response::new()
+        .add_attribute("method", "instantiate")
+        .add_attribute("admin", admin))
+}
```

Comment:
UX note. It's much easier for a DAO to instantiate and fund a contract like this in one prop. Otherwise, DAOs will have to vote to instantiate the contract and then also fund it.

This will only work with native tokens, but we're moving to Token Factory anyway.

Reply:
can you share more about Token Factory)